runtime/CSharp/src/Tree/Trees.cs

```diff
@@ -36,9 +37,7 @@ public static string ToStringTree(ITree t)
         /// </remarks>
         public static string ToStringTree(ITree t, Parser recog)
         {
-            string[] ruleNames = recog != null ? recog.RuleNames : null;
-            IList<string> ruleNamesList = ruleNames != null ? Arrays.AsList(ruleNames) : null;
-            return ToStringTree(t, ruleNamesList);
+            return ToStringTree(t, recog?.RuleNames);
```

Comment:
Do we have a test proving this still works ?

Reply:
We don't have, but it's trivial rewriting. `AsList` method just does nothing: https://github.com/antlr/antlr4/pull/3861/files#diff-a5b1c1aea0d0e2b922a00605b9181ebdf38ea00cef8a58464bfa777f0c51d8beL22

Reply:
It looks like `AsList` was added for imitation of Java code. But C# is a different language, and the same code can be written in a more simple way.

Reply:
Java does not have a ? operator, that's my concern

Reply:
I don't trust reasoning, that's why I'm always requiring tests